dev/src/bulk-writer.ts

```diff
@@ -648,14 +789,20 @@ export class BulkWriter {
    *   console.log('Executed all writes');
    * });
    */
-  close(): Promise<void> {
+  async close(): Promise<void> {
     this.verifyNotClosed();
     this.firestore._decrementBulkWritersCount();
     const flushPromise = this.flush();
     this.closed = true;
-    return flushPromise;
+    await flushPromise;
+
+    // Make sure user promises resolve before close() resolves.
+    await Promise.all(this._pendingOps);
```

Comment:
I think the exact problem that this line addresses also exists in `flush` (i.e. a retry can take longer to resolve than the Promise returned from this method). We should move this logic to `_flush`.